src/handlers.rs

```diff
@@ -979,6 +979,92 @@ pub fn get_banned_public_keys(
     return Ok(warp::reply::json(&response).into_response());
 }
 
+// Pinned Messages
+pub fn pin_message(
+    id: i64, auth_token: &str, pool: &storage::DatabaseConnectionPool,
+) -> Result<Response, Rejection> {
+    // Check authorization level
+    let (has_authorization_level, requesting_public_key) =
+        has_authorization_level(auth_token, AuthorizationLevel::Basic, pool)?;
+    if !has_authorization_level {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    if !is_moderator(&requesting_public_key, pool)? {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    // Get a connection and open a transaction
+    let mut conn = pool.get().map_err(|_| Error::DatabaseFailedInternally)?;
+    let tx = conn.transaction().map_err(|_| Error::DatabaseFailedInternally)?;
+    // Pin the message if it's present
+    let stmt = format!("UPDATE MAIN_TABLE SET pinned_message_id = (?1)");
```

Comment:
`MAIN_TABLE` was just the name of the constant, the actual table name is `main` (and same for all the other dropped table constants).  Also no need for `format` these anymore since there's no pattern in them.
```suggestion
    let stmt = "UPDATE main SET pinned_message_id = ?";
```